pkg/jobs/adopt.go

```diff
@@ -423,18 +424,53 @@ func (r *Registry) runJob(
 	return err
 }
 
+const clearClaimQuery = `
+   UPDATE system.jobs
+      SET claim_session_id = NULL, claim_instance_id = NULL
+    WHERE id = $1
+      AND claim_session_id = $2
+      AND claim_instance_id = $3
+      AND status NOT IN ('` + string(StatusPauseRequested) + `', '` + string(StatusCancelRequested) + `')`
+
+// maybeClearLease clears the claim on the given job, provided that
+// the current lease matches our liveness Session.
+func (r *Registry) maybeClearLease(job *Job, jobErr error) {
+	if jobErr == nil {
+		return
+	}
+
+	// We use the serverCtx here rather than the context from the
+	// caller since the caller's context may have been canceled.
+	r.withSession(r.serverCtx, func(ctx context.Context, s sqlliveness.Session) {
+		err := r.db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
```

Comment:
I think we'd be better off not using a txn loop if we can avoid it. This is a txn which can use parallel commits. I'd rather that it did that by using a nil txn passed to the ie. I think if you want low priority you maybe can do that through session data.

Reply:
I've updated this and decided to go forward without a low priority query for now.  

Do you happen to know if there is a good way to verify that the query is in fact using parralel commit?

Reply:
A KV trace would show you. Not exactly sure how to get that in this context. I think maybe you could install something into sessiondata and then the context?

Like, if you did:

```
root@localhost:26257/defaultdb> create table foo (i int primary key, j int, index(j));
CREATE TABLE


Time: 897ms total (execution 897ms / network 0ms)

root@localhost:26257/defaultdb> set tracing = kv;
SET TRACING


Time: 0ms total (execution 0ms / network 0ms)

root@localhost:26257/defaultdb> insert into foo values (5, 1);
INSERT 0 1


Time: 45ms total (execution 44ms / network 0ms)

root@localhost:26257/defaultdb> show kv  trace for session;
            timestamp           |       age       |                     message                      |                        tag                         |                location                 |              operation               | span
--------------------------------+-----------------+--------------------------------------------------+----------------------------------------------------+-----------------------------------------+--------------------------------------+-------
  2022-11-10 15:35:54.82937+00  | 00:00:00.003108 | rows affected: 1                                 | [n1,client=127.0.0.1:54700,user=root]              | sql/conn_executor_exec.go:723           | sql query                            |    2
  2022-11-10 15:35:59.56418+00  | 00:00:04.737918 | CPut /Table/105/1/5/0 -> /TUPLE/2:2:Int/1        | [n1,client=127.0.0.1:54700,user=root]              | sql/row/writer.go:215                   | count                                |   18
  2022-11-10 15:35:59.564228+00 | 00:00:04.737966 | InitPut /Table/105/2/1/5/0 -> /BYTES/            | [n1,client=127.0.0.1:54700,user=root]              | sql/row/inserter.go:183                 | count                                |   18
  2022-11-10 15:35:59.564313+00 | 00:00:04.738051 | querying next range at /Table/105/1/5/0          | [n1,client=127.0.0.1:54700,user=root,txn=5ad5ea61] | kv/kvclient/kvcoord/range_iter.go:182   | dist sender send                     |   20
  2022-11-10 15:35:59.564371+00 | 00:00:04.738109 | querying next range at /Table/105/2/1/5/0        | [n1,client=127.0.0.1:54700,user=root,txn=5ad5ea61] | kv/kvclient/kvcoord/range_iter.go:182   | dist sender send                     |   20
  2022-11-10 15:35:59.564409+00 | 00:00:04.738147 | r58: sending batch 1 InitPut to (n1,s1):1        | [n1,client=127.0.0.1:54700,user=root,txn=5ad5ea61] | kv/kvclient/kvcoord/dist_sender.go:2048 | dist sender send                     |   20
  2022-11-10 15:35:59.564476+00 | 00:00:04.738214 | r57: sending batch 1 CPut, 1 EndTxn to (n1,s1):1 | [n1,client=127.0.0.1:54700,user=root,txn=5ad5ea61] | kv/kvclient/kvcoord/dist_sender.go:2048 | kv.DistSender: sending partial batch |   21
  2022-11-10 15:35:59.606589+00 | 00:00:04.780327 | fast path completed                              | [n1,client=127.0.0.1:54700,user=root]              | sql/plan_node_to_row_source.go:176      | count                                |   18
  2022-11-10 15:35:59.607296+00 | 00:00:04.781034 | rows affected: 1                                 | [n1,client=127.0.0.1:54700,user=root]              | sql/conn_executor_exec.go:723           | sql query                            |   12
  2022-11-10 15:35:59.610933+00 | 00:00:04.784671 | rows affected: 1                                 | [n1,client=127.0.0.1:54700,user=root]              | sql/conn_executor_exec.go:723           | sql query                            |   27
(10 rows)
```

That's definitely a parallel commit.